packages/sui-domain/src/fetcher/Fetcher.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @interface
+ * */
+export default class Fetcher {
```

Comment:
As we use the "required" helper, it could be a good idea to do it with interfaces as well.

```js
const getMethods = obj => Object.getOwnPropertyNames(obj)
  .filter(key => typeof obj[key] == 'function')

const getClassName = obj => obj.name || `${obj.constructor.name}.prototype`

const compareMethods = compared => base => getMethods(base)
    .forEach((key) => {
      if(!compared[key] || base[key] === compared[key]) {
        throw new Error(`${getClassName(base)}.${key}() must be implemented.`)
      }
    })
   
const impl = instance => Class => {
   compareMethods(instance)(Class.prototype)
   compareMethods(instance.__proto__.constructor)(Class)  
}

export default impl
```

```js

import impl from './implements'
class Fetcher {
  constructor() {
     impl(this)(Fetcher)
  }
}
```

Reply:
I would like to use Flow for this instead of having to put in every constructor we use. Even if we do not do this kind of check in production it feels we're going to far with that.